src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -709,6 +817,113 @@ biclique xs ys = let sxs = Set.fromList xs
                   in BAM (Map.fromSet (const sys) sxs)
                          (Map.fromSet (const sxs) sys)
 
+-- | The /star/ formed by a center vertex connected to a list of leaves.
+-- Complexity: /O(L log(L))/ time, where /L/ is the length of the given list.
+--
+-- @
+-- star x []     == 'leftVertex' x
+-- star x [y]    == 'edge' x y
+-- star x [x]    == 'edge' x x
+-- star x [y, z] == 'edges' [(x, y), (x, z)]
+-- star x ys     == 'connect' ('leftVertex' x) ('vertices' [] ys)
+-- @
+star :: (Ord a, Ord b) => a -> [b] -> AdjacencyMap a b
+star x ys = overlay (leftVertex x) (edges [ (x, y) | y <- ys ])
+
+-- | The /stars/ formed by overlaying a list of 'star's. An inverse of
+-- 'leftAdjacencyList'.
+-- Complexity: /O(L log(L))/ time, where /L/ is the total size of the input.
+--
+-- @
+-- stars []                      == 'empty'
+-- stars [(x, [])]               == 'leftVertex' x
+-- stars [(x, [y])]              == 'edge' x y
+-- stars [(x, ys)]               == 'star' x ys
+-- stars                         == 'overlays' . 'map' ('uncurry' 'star')
+-- 'overlay' (stars xs) (stars ys) == stars (xs ++ ys)
+-- @
+stars :: (Ord a, Ord b) => [(a, [b])] -> AdjacencyMap a b
+stars = overlays . map (uncurry star)
+
+-- | Construct a /mesh/ graph from two lists of vertices.
+-- Complexity: /O(L1 * L2 + L1 log(L1) + L2 log(L2))/ time, where /L1/ and
+-- /L2/ are the lengths of the given lists.
+-- 
+-- @
+-- mesh xs []             == 'empty'
+-- mesh [] ys             == 'empty'
+-- mesh [x] [y]           == 'leftVertex' (x, y)
+-- mesh [1, 2] [\'a\', \'b\'] == 'biclique' [(1, \'a\'), (2, \'b\')] [(1, \'b\'), (2, \'a\')]
+-- mesh [1, 1] [\'a\', \'b\'] == 'biclique' [(1, \'a\'), (1, \'b\')] [(1, \'a\'), (1, \'b\')]
+-- @
+mesh :: (Ord a, Ord b) => [a] -> [b] -> AdjacencyMap (a, b) (a, b)
+mesh xs ys = box (path (fromList xs)) (path (fromList ys))
+
+-- | Compute the /Cartesian product/ of bipartite graphs. The vertices are
+-- sorted into parts using given combinators. Values returned by combinators
+-- are not necessarily distinct.
+-- Complexity: /O(s1 * s2 * log(s1 * s2))/ time and /O(s1 * s2)/ memory, where
+-- /s1/ and /s2/ are the sizes of the given graphs.
+--
+-- There's also a shorthand 'box' for @boxWith (,) (,) (,) (,)@.
+--
+-- @
+-- 'box' ('path' [0,1]) ('path' [\'a\',\'b\']) == 'edges' [ ((0,\'a\'), (0,\'b\'))
```

Comment:
It's strange to have `box` examples attached to a different function (`boxWith`) -- I suggest simply moving these examples to `box`, and adjusting the comments accordingly. Also, move `box` in front of more complicated `boxWith` both here and in the export list.